geps/gep-1619.md

```diff
@@ -130,98 +275,604 @@ Generally, the implementation API programs the dataplane API; however these two
+of a service after the gateway has established persistent sessions with it.
+
+```yaml
+kind: Service
+metadata:
+  name: my-service
+spec:
+  selector:
+    app.kubernetes.io/name: MyApp # Service selector can change
+```
+
+The expected behavior is that the gateway will retain existing persistent sessions, even if the pod is no longer
+selected, and establish new persistent sessions after a selector update. This use case is uncommon and may not be
+supported by some implementations due to their current designs.
 
 ### Open Questions
 
-- Should we include both session persistence and session affinity API configurations in this GEP or just focus on session persistence?
-- Should both session persistence and session affinity be designed as common load balancing policy API? Should they be configured as separately in the API?
 - What happens when session persistence is broken because the backend is not up or healthy? If that's an error case, how should that be handled? Should the API dictate the http error code? Or should the API dictate fall back behavior?
+- What happens when session persistence causes traffic splitting scenarios to overload a backend?
+- Should we add status somewhere when a user gets in to a "risky" configuration with session persistence?
+- Should there be an API configuration field that specifies how already established sessions are handled?
 
 ## TODO
 The following are items that we intend to resolve before we consider this GEP implementable:
 
-- We need to identify the needs and use cases from end users for specifying session persistence.
 - We need to identify and document requirements regarding session draining and migration. How do implementations drain established sessions during backend upgrades without disruption?
+  - Do we need a "session draining timeout" as documented by [A55: xDS-Based Stateful Session Affinity for Proxyless gRPC](https://github.com/grpc/proposal/blob/master/A55-xds-stateful-session-affinity.md#background)
+    define in this API?
 - We need to document sessions with Java in greater detail. Java standardized the API and behavior of session persistence long ago and would be worth examining.
 - We need to add a small section on compliance regarding the browser and client relationship.
-- We need to design and document an API for session persistence and/or session affinity in the Gateway API spec (see [open question](#open-questions) about API design)
+- We need to finish enumerating all the edge cases in [Expected API Behavior](#expected-api-behavior) and identify
+potential scenarios where session persistence could break so implementation can implement session persistence in a
+predicable way.
+- Update the GEP for targeting a route rule section when [#2283](https://github.com/kubernetes-sigs/gateway-api/pull/2283)
+gets merged in [GEP-713](/geps/gep-713).
```

Comment:
#2283 is already merged, so we can remove the bullet and update the rest of this doc to reflect GEP-713. BTW, GEP-713 is "experimental" so there is no reason to have conditional or contingent clauses about targeting route rule section in my opinion.

Reply:
Right - done.